Biasing/src/Biasing/TargetProcessFilter.cxx

```diff
@@ -50,19 +50,38 @@ void TargetProcessFilter::stepping(const G4Step* step) {
   // Get the track associated with this step.
   auto track{step->GetTrack()};
 
+  if (G4EventManager::GetEventManager()->GetConstCurrentEvent()->IsAborted())
+    return;
+
   // Get the track info and check if this track is a brem candidate
   auto trackInfo{simcore::UserTrackInformation::get(track)};
   if ((trackInfo != nullptr) && !trackInfo->isBremCandidate()) return;
 
-  // Get the region the particle is currently in.  Continue processing
-  // the particle only if it's in the calorimeter region.
+  // Get the particles daughters.
+  auto secondaries{step->GetSecondary()};
+
+  // Get the region the particle is currently in. Continue processing
+  // the particle only if it's in the target region.
   if (auto region{
           track->GetVolume()->GetLogicalVolume()->GetRegion()->GetName()};
-      region.compareTo("target") != 0)
+      region.compareTo("target") != 0) {
+    // If secondaries were produced outside of the volume of interest,
+    // and there aren't additional brems to process, abort the event.
+    // Otherwise, suspend the track and move on to the next brem.
+    if (secondaries->size() != 0) {
+      if (getEventInfo()->bremCandidateCount() == 1) {
+	track->SetTrackStatus(fKillTrackAndSecondaries);
+	G4RunManager::GetRunManager()->AbortEvent();
+	currentTrack_ = nullptr;
+      } else {
+	currentTrack_ = track;
+	track->SetTrackStatus(fSuspend);
+	getEventInfo()->decBremCandidateCount();
+	trackInfo->tagBremCandidate(false);
+      }
+    }
```

Comment:
It looks like your editor used tabs instead of spaces here, no bother - it is an easy fix.

```suggestion
    if (secondaries->size() != 0) {
      if (getEventInfo()->bremCandidateCount() == 1) {
        track->SetTrackStatus(fKillTrackAndSecondaries);
        G4RunManager::GetRunManager()->AbortEvent();
        currentTrack_ = nullptr;
      } else {
      currentTrack_ = track;
      track->SetTrackStatus(fSuspend);
      getEventInfo()->decBremCandidateCount();
      trackInfo->tagBremCandidate(false);
      }
    }
```